main/src/com/google/refine/importing/UrlRewriter.java

```diff
@@ -33,12 +33,32 @@
 
 package com.google.refine.importing;
 
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.List;
+
 public interface UrlRewriter {
     static public class Result {
         public String rewrittenUrl;
         public String format;
         public boolean download;
+        public String metaDataFormat;
+        
+        public Result(String rewrittenUrl, String format, boolean download) {
+            this.rewrittenUrl = rewrittenUrl;
+            this.format = format;
+            this.download = download;
+        }
+        
+        public Result(String rewrittenUrl, String format, boolean download, String metaDataFormat) {
+            this.rewrittenUrl = rewrittenUrl;
+            this.format = format;
+            this.download = download;
+            this.metaDataFormat = metaDataFormat;
+        }
     }
     
-    public Result rewrite(String url);
+    public List<Result> rewrite(String url) throws MalformedURLException, IOException;
+    
+    public boolean filter(String url);
 }
```

Comment:
It looks like this interface is used to translate relative paths to full URLs - maybe it would be worth putting a few comments explaining what it does (for instance, why does it return a List of Results instead of just a Result)?

Reply:
comment added